### DISCUSSION TITLE
Support versioning for iOS plist

### DISCUSSION THREAD

**Discussion by KG3RK3N** [rfennell/AzurePipelines#1156] [General]
It's planned to support iOS plist files for the values `CFBundleShortVersionString` and `CFBundleVersion` ?

**Comment by rfennell** [comment]
Sorry, what task are you talking about?

**Reply by KG3RK3N** [reply]
I talk about a task like the android manifest versioning task but for iOS and its plist file.

**Reply by rfennell** [reply]
I did not have a plan to support this file format, but only because nobody had asked.

After a quick bit of research I see it is an XML format, so could you not use my generic [update-xml-file](https://github.com/rfennell/AzurePipelines/wiki/File-Copier-Tasks#update-xml-file)?

I am not a iOS developer, if this task won't do the job, could you provide a sample `plist` file so test against.

**Reply by KG3RK3N** [reply]
ah okay, i've don't see that you have a general xml task.

Here a example of a plist file:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
  <dict>
    <key>CFBundleDevelopmentRegion</key>
    <string>en</string>
    <key>CFBundleDisplayName</key>
    <string>${PRODUCT_NAME}</string>
    <key>CFBundleExecutable</key>
    <string>${EXECUTABLE_NAME}</string>
    <key>CFBundleInfoDictionaryVersion</key>
    <string>6.0</string>
    <key>CFBundleName</key>
    <string>${PRODUCT_NAME}</string>
    <key>CFBundlePackageType</key>
    <string>APPL</string>
    <key>CFBundleShortVersionString</key>
    <string>1.0</string>
    <key>CFBundleSignature</key>
    <string>????</string>
    <key>CFBundleVersion</key>
    <string>1.0</string>
    <key>LSRequiresIPhoneOS</key>
    <true/>
    <key>UILaunchStoryboardName</key>
    <string>LaunchScreen</string>
    <key>UIRequiresFullScreen</key>
    <true/>
    <key>UIRequiredDeviceCapabilities</key>
    <array>
      <string>armv7</string>
    </array>
    <key>UISupportedInterfaceOrientations</key>
    <array>
      <string>UIInterfaceOrientationPortrait</string>
      <string>UIInterfaceOrientationLandscapeLeft</string>
      <string>UIInterfaceOrientationLandscapeRight</string>
    </array>
    <key>UISupportedInterfaceOrientations~ipad</key>
    <array>
      <string>UIInterfaceOrientationPortrait</string>
      <string>UIInterfaceOrientationPortraitUpsideDown</string>
      <string>UIInterfaceOrientationLandscapeLeft</string>
      <string>UIInterfaceOrientationLandscapeRight</string>
    </array>
    <key>UIStatusBarStyle</key>
    <string>UIStatusBarStyleLightContent</string>
    <key>UIViewControllerBasedStatusBarAppearance</key>
    <false/>
  </dict>
</plist>
```

Currently I'm not sure if that works so easy with the general xml task because you can see that the key and the value are not referenced directly. 
In my opinion with the general xml task I must get the index of the element with the key and increment that to get the value for the right key. If that possible, then it should works.

**Reply by rfennell** [reply]
The generic task is hidden in a different extension, so easy to miss 😃

Thanks for the sample, seems a good candidate for a custom task, I will add it to the backlog

**Comment by rfennell** [comment]
Shipped in 2.33.16
